dumpling/export/sql.go

```diff
@@ -244,50 +245,65 @@ func ListAllDatabasesTables(tctx *tcontext.Context, db *sql.Conn, databaseNames
 					return errors.Trace(err2)
 				}
 				avgRowLength = 0 // can't get avgRowLength from the result of `show full tables` so hardcode to 0 here
-				dbTables[schema] = append(dbTables[schema], &TableInfo{table, avgRowLength, tableType})
+				dbTables[schema] = append(dbTables[schema], &TableInfo{table, avgRowLength, tableType, ""})
 				return nil
 			}, query); err != nil {
 				return nil, errors.Annotatef(err, "sql: %s", query)
 			}
 		}
-	default:
-		const queryTemplate = "SHOW TABLE STATUS FROM `%s`"
-		selectedTableType := make(map[TableType]struct{})
-		for _, tableType = range tableTypes {
-			selectedTableType[tableType] = struct{}{}
+	}
+
+	const queryTemplate = "SHOW TABLE STATUS FROM `%s`"
```

Comment:
Seems only `listTableByShowFullTables ` mode need this extra step

Reply:
there has `listTableByShowTableStatus` by default.